lib/ansible/modules/cloud/azure/azure_rm_acr.py

```diff
@@ -0,0 +1,362 @@
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        to_do = Actions.NoAction
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        # Check if the ACR instance already present in the RG
+        if self.state == 'present':
+            response = self.get_acr()
+            self.results['state'] = response
+
+            if not response:
+                to_do = Actions.Create
+            else:
+                self.log('Results : {0}'.format(response))
+                if response['provisioning_state'] == "Succeeded":
+                    to_do = Actions.Update
+                else:
+                    to_do = Actions.NoAction
+
+            self.log("Create / Update the ACR instance")
+            if self.check_mode:
+                return self.results
+
+            self.results['state'] = self.create_update_acr(to_do)
+            if to_do != Actions.NoAction:
+                self.results['changed'] = True
+            else:
+                self.results['changed'] = False
+
+            self.log("ACR instance created or updated")
+        elif self.state == 'absent':
+            if self.check_mode:
+                return self.results
+            self.delete_acr()
+            self.log("ACR instance deleted")
+
+        return self.results
+
+    def create_update_acr(self, to_do):
+        '''
+        Creates or updates a container registry.
+
+        :return: deserialized ACR instance state dictionary
+        '''
+        self.log("Creating / Updating the ACR instance {0}".format(self.name))
+
+        try:
+            if to_do != Actions.NoAction:
+                if to_do == Actions.Create:
+                    name_status = self.containerregistry_mgmt_client.registries.check_name_availability(self.name)
+                    if name_status.name_available:
+                        poller = self.containerregistry_mgmt_client.registries.create(
+                            resource_group_name=self.resource_group,
+                            registry_name=self.name,
+                            registry=Registry(
+                                location=self.location,
+                                sku=Sku(
+                                    name=self.sku
+                                ),
+                                tags=self.tags,
+                                admin_user_enabled=self.admin_user_enabled
+                            )
+                        )
+                    else:
+                        raise Exception("Invalid registry name. reason: " + name_status.reason + " message: " + name_status.message)
+                else:
+                    registry = self.containerregistry_mgmt_client.registries.get(self.resource_group, self.name)
```

Comment:
Not understanding why we need the intermediate `get` here for update- we wouldn't be in the update state if the registry didn't already exist in this run of the module, so this seems superfluous.